addons/stock_account/models/product.py

```diff
@@ -39,18 +39,23 @@ class ProductTemplate(models.Model):
         help="When doing real-time inventory valuation, counterpart journal items for all outgoing stock moves will be posted in this account, unless "
              "there is a specific valuation account set on the destination location. When not set on the product, the one from the product category is used.")
     average_price = fields.Float(
-        'Average Cost', compute='_compute_average_price',
+        'Average Cost', compute='_compute_fifo_average_price',
         digits=dp.get_precision('Product Price'), groups="base.group_user",
-        help="Average cost of the product, in the default unit of measure of the product.")
+        help="This is the average cost of this product and can be used to value the stock of this product. ")
+    fifo_price = fields.Float(
+        'FIFO Cost', compute='_compute_fifo_average_price',
+        digits=dp.get_precision('Product Price'), groups="base.group_user",
+        help="This is the cost based on the last products leaving the stock.")
```

Comment:
This means that if you just received the product and did not deipatch any sale you can not have a "Fifo BAsed cost"?.

At least what I understand from the help text.

I want to point that the very first cost of the product is generally the one used to set the sale price.